mu4e/meson.build

```diff
@@ -89,6 +89,21 @@ foreach src : mu4e_srcs
                           '--funcall', 'batch-byte-compile', '@INPUT@'])
 endforeach
 
+if get_option('native-comp')
+  target_func = '(setq native-comp-eln-load-path (list "' + mu4e_nativelispdir + '"))'
+  custom_target('native-comp',
+                build_by_default: true,
+                input: mu4e_srcs,
+                output: 'native-comp',
```

Comment:
This produces an output file named literally "native-comp"? Seems odd.

What's the goal here, to produce compiled outputs in the build directory or to have them be installed? What happens if the user doesn't have root access during build, only during install? What happens if someone wants to use this but in conjunction with `$DESTDIR`?

Reply:
> This produces an output file named literally "native-comp"?

No, output is not used here. I specify it because it is mandatory by meson syntax. The output eln files are directly put into mu4e_nativelispdir, which is named `native-lisp` and  is in the same dir as mu4e_lispdirdir, by emacs.

> What happens if the user doesn't have root access during build, only during install?

I only consider the usecase of installing using a package manager, have not considered manually building and installing. I have test it with nix, a system package manager like apt, and it works well. I think I have merged building and installing, will try to learn more meson and work on that later.

> What happens if someone wants to use this but in conjunction with $DESTDIR

I do not think this will be an issue since eln files are installed together with elc files. Assume elc files are installed correctly in this case, so do eln files.

Reply:
Since my meson knowledge is limited, guides on how to seperate the building phase and installation phase is welcome.

Reply:
On splitting build and install phases, I can put build results in a temp dir. My question is how to move or copy a dir in install phase with custom_target.

I do not think we can specify the file name of an eln file.

Reply:
> I only consider the usecase of installing using a package manager, have not considered manually building and installing.

No, you have considered a variety of use cases *other* than a package manager. :)

> I have test it with nix, a system package manager like apt

Nix is not a package manager like apt. Nix is a package manager unique for not doing things like any other Linux package manager (I am one of the major authors of a Linux package manager, I know how this works).

> I do not think this will be an issue since eln files are installed together with elc files. Assume elc files are installed correctly in this case, so do eln files.

Thereby demonstrating exactly why Nix is not a package manager like the usual ones (the ones I'm concerned about).

Nix doesn't package to DESTDIR, it packages directly to "manually building and installing into a custom --prefix".

That is the *only* reason why it works for Nix.

> since eln files are installed together with elc files

They very explicitly are not. https://github.com/djcb/mu/blob/4cceded947e9924b880ea35d79cec1095022f210/mu4e/meson.build#L71-L90

This (elc handling) is *extremely* different from your PR to do eln handling. The elc files are written into the ***build*** directory, the eln files are proposed in this PR to be written to the ***install*** directory.

It is of the utmost importance, regardless of build system (it has *nothing* to do with Meson) that files are *compiled* and produced in the build directory, and only installed by a separate install step. If it works correctly in that case (the case of "basically all Linux distros"), then it will work correctly for Nix as well, but the reverse is NOT true.

Reply:
> What happens if the user doesn't have root access during build, only during install? What happens if someone wants to use this but in conjunction with $DESTDIR?

I have separated the build and install phases. As a result, your initial concerns should all be addressed.